tests/test_sign.py

```diff
@@ -1,89 +1,180 @@
+        instructions.confirm_transaction()
+    return instructions
 
-def test_untrusted_hash_sign_fail_short_payload(cmd, transport):
-    # should fail if the payload is less than 7 bytes
-    sw, _ = transport.exchange(0xE0, 0x48, 0x00, 0x00, None, b"\x01\x02\x03\x04\x05\x06")
-    assert sw == 0x6700
+def sign_psbt_instruction_approve_2(model: Firmware) -> Instructions:
+    instructions = Instructions(model)
 
+    if model.name.startswith("nano"):
+        instructions.new_request("Accept")
+        instructions.new_request("Accept")
+    else:
+        instructions.navigate_end_of_flow()
+        instructions.confirm_transaction()
+    return instructions
 
-@automation("automations/accept.json")
-def test_sign_p2wpkh_accept(cmd):
-    for filepath in Path("data").rglob("p2wpkh/tx.json"):
-        sign_from_json(cmd, filepath)
+def open_psbt_from_file(filename: str) -> PSBT:
+    raw_psbt_base64 = open(filename, "r").read()
 
+    psbt = PSBT()
+    psbt.deserialize(raw_psbt_base64)
+    return psbt
 
-@automation("automations/accept.json")
-def test_sign_p2sh_p2wpkh_accept(cmd):
-    for filepath in Path("data").rglob("p2sh-p2wpkh/tx.json"):
-        sign_from_json(cmd, filepath)
 
+def test_sign_psbt_singlesig_pkh_1to1(navigator: Navigator, firmware: Firmware, client: RaggerClient, test_name: str):
 
-@automation("automations/accept.json")
-def test_sign_p2pkh_accept(cmd):
-    for filepath in Path("data").rglob("p2pkh/tx.json"):
-        sign_from_json(cmd, filepath)
+    # PSBT for a legacy 1-input 1-output spend (no change address)
+    psbt = open_psbt_from_file(f"{tests_root}/psbt/singlesig/pkh-1to1.psbt")
 
+    wallet = WalletPolicy(
+        "",
+        "pkh(@0/**)",
+        [
+            "[f5acc2fd/44'/1'/0']tpubDCwYjpDhUdPGP5rS3wgNg13mTrrjBuG8V9VpWbyptX6TRPbNoZVXsoVUSkCjmQ8jJycjuDKBb9eataSymXakTTaGifxR6kmVsfFehH1ZgJT"
+        ],
+    )
 
-@automation("automations/reject.json")
-def test_sign_fail_p2pkh_reject(cmd):
-    with pytest.raises(ConditionOfUseNotSatisfiedError):
-        sign_from_json(cmd, "./data/one-to-one/p2pkh/tx.json")
\ No newline at end of file
+    # expected sigs:
+    # #0:
+    #  "pubkey" : "02ee8608207e21028426f69e76447d7e3d5e077049f5e683c3136c2314762a4718",
+    #  "signature" : "3045022100e55b3ca788721aae8def2eadff710e524ffe8c9dec1764fdaa89584f9726e196022012a30fbcf9e1a24df31a1010356b794ab8de438b4250684757ed5772402540f401"
+    result = client.sign_psbt(psbt, wallet, None, navigator=navigator,
+                              instructions=sign_psbt_instruction_approve(firmware),
+                              testname=test_name)
+
+    assert result == [(
+        0,
+        PartialSignature(
+            pubkey=bytes.fromhex("02ee8608207e21028426f69e76447d7e3d5e077049f5e683c3136c2314762a4718"),
+            signature=bytes.fromhex(
+                "3045022100e55b3ca788721aae8def2eadff710e524ffe8c9dec1764fdaa89584f9726e196022012a30fbcf9e1a24df31a1010356b794ab8de438b4250684757ed5772402540f401"
```

Comment:
Same, signature and pubkey could be validated without hardcoding the result I guess?